src/main/java/org/terasology/climateConditions/HyperthermiaSystem.java

```diff
@@ -64,29 +59,50 @@ public void onBiomeChange(OnBiomeChangedEvent event, EntityRef player) {
         }
     }
 
-    @ReceiveEvent(components = {HyperthermiaComponent.class})
-    public void modifySpeed(GetMaxSpeedEvent event, EntityRef player) {
-        event.multiply(walkSpeedMultiplier);
+    /**
+     * Reduces the walking/running speed of the player.
+     * Is only active iff the player has a {@link HyperthermiaComponent}.
+     */
+    @ReceiveEvent
+    public void modifySpeed(GetMaxSpeedEvent event, EntityRef player, HyperthermiaComponent hyperthermia) {
+        event.multiply(hyperthermia.walkSpeedMultiplier);
     }
 
-    @ReceiveEvent(components = {HyperthermiaComponent.class})
-    public void modifyJumpSpeed(AffectJumpForceEvent event, EntityRef player) {
-        event.multiply(jumpSpeedMultiplier);
+    /**
+     * Reduces the jump speed of the player.
+     * Is only active iff the player has a {@link HyperthermiaComponent}.
+     */
+    @ReceiveEvent
+    public void modifyJumpSpeed(AffectJumpForceEvent event, EntityRef player, HyperthermiaComponent hyperthermia) {
+        event.multiply(hyperthermia.jumpSpeedMultiplier);
     }
 
-    @ReceiveEvent(components = {HyperthermiaComponent.class})
-    public void onHyperthermia(OnAddedComponent event, EntityRef player, HealthComponent health) {
-        weakenPlayer(player, health);
+    /**
+     * Weakens the player by reducing the maxHealth of the player when {@link HyperthermiaComponent} is added.
+     */
+    @ReceiveEvent
+    public void onHyperthermia(OnAddedComponent event, EntityRef player, HealthComponent health,
+                               HyperthermiaComponent hyperthermia) {
+        weakenPlayer(player, health, hyperthermia);
     }
 
-    @ReceiveEvent(components = {HyperthermiaComponent.class})
-    public void beforeRemoveHyperthermia(BeforeRemoveComponent event, EntityRef player, HealthComponent health) {
-        removePlayerWeakness(player, health);
+    /**
+     * Removes the weakness of the player by restoring the maxHealth of the player to the original value when {@link
```

Comment:
This sounds like it should be the documentation for `removePlayerWeakness` instead of `beforeRemoveHyperthermia`